src/ray/rpc/gcs_server/gcs_rpc_client.h

```diff
@@ -98,8 +98,8 @@ class Executor {
         callback(status, reply);                                                       \
         delete executor;                                                               \
       } else {                                                                         \
-        gcs_service_failure_detected_(GcsServiceFailureType::RPC_DISCONNECT);          \
-        executor->Retry();                                                             \
+        gcs_service_failure_detected_(GcsServiceFailureType::RPC_DISCONNECT,           \
+                                      [executor]() { executor->Retry(); });            \
```

Comment:
When is executor deallocated?

Reply:
It doesn't affect the lifecycle of `executor`, deallocated when the rpc call is timed out or success.